nym-validator-rewarder/src/rewarder/storage/mod.rs

```diff
@@ -79,89 +104,95 @@ impl RewarderStorage {
-                        epoch_id,
-                        validator.validator.consensus_address,
-                        validator.operator_account.to_string(),
-                        validator.whitelisted,
-                        reward_amount,
-                        validator.voting_power_at_epoch_start,
-                        validator.voting_power_ratio.to_string(),
-                        validator.signed_blocks,
-                        validator.ratio_signed.to_string(),
-                    )
-                    .await?;
+        // block signing info
+        if let Ok(block_signing) = reward.signing {
+            self.manager
+                .insert_rewarding_epoch_block_signing(
+                    epoch_id,
+                    block_signing
+                        .as_ref()
+                        .map(|s| s.total_voting_power_at_epoch_start)
+                        .unwrap_or_default(),
+                    block_signing.as_ref().map(|s| s.blocks).unwrap_or_default(),
+                    reward.signing_budget.to_string(),
+                )
+                .await?;
+            if let Some(signing) = block_signing {
+                for validator in signing.validators {
+                    let reward_amount = validator.reward_amount(&reward.signing_budget).to_string();
+                    self.manager
+                        .insert_rewarding_epoch_block_signing_reward(
+                            epoch_id,
+                            validator.validator.consensus_address,
+                            validator.operator_account.to_string(),
+                            validator.whitelisted,
+                            reward_amount,
+                            validator.voting_power_at_epoch_start,
+                            validator.voting_power_ratio.to_string(),
+                            validator.signed_blocks,
+                            validator.ratio_signed.to_string(),
+                        )
+                        .await?;
+                }
             }
+        } else {
+            self.insert_failed_rewarding_epoch_block_signing(epoch_id, &reward.signing_budget)
+                .await?;
         }
 
-        // safety: we must have at least a single value here
-        #[allow(clippy::unwrap_used)]
-        let dkg_epoch_start = reward
-            .credentials
-            .as_ref()
-            .and_then(|c| c.dkg_epochs.first().copied())
-            .unwrap_or_default();
-        #[allow(clippy::unwrap_used)]
-        let dkg_epoch_end = reward
-            .credentials
-            .as_ref()
-            .and_then(|c| c.dkg_epochs.last().copied())
-            .unwrap_or_default();
-
-        self.manager
-            .insert_rewarding_epoch_credential_issuance(
+        // credential info
+        if let Ok(credential_issuance) = reward.credentials {
+            // safety: we must have at least a single value here
+            #[allow(clippy::unwrap_used)]
```

Comment:
is this needed?

Reply:
technically no, but in ideal world we should have global 
```
#![warn(clippy::expect_used)]
#![warn(clippy::unwrap_used)]
```
everywhere and this will silence the future warnings in that particular location